.github/workflows/master-build.yml

```diff
@@ -0,0 +1,51 @@
+name: Build and merge-publish
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  pull_request:
+    branches:
+      - main
+    paths-ignore:
+      - '**/*.md'
+  create:
+  pull_request_target:
+
+jobs:
+  build-and-test-code:
+    runs-on: ubuntu-20.04
+    steps:
+      # Set fetch-depth: 0 to fetch all history for all branches and tags.
+      - name: Check out code
+        uses: actions/checkout@v2.3.4
+        with:
+          fetch-depth: 2
+      
+      - name: create checksum file
+        run: find . -type f -name "*.gradle*" -o -name "gradle-wrapper*" | xargs shasum | sort > checksum.txt && cat checksum.txt
+
+      - name: Cache packages
+        id: cache-packages
+        uses: actions/cache@v2
+        with:
+          path: ~/.gradle
+          key: gradle-packages-${{ runner.os }}-${{ hashFiles('**/checksum.txt') }}
+          restore-keys: gradle-packages-${{ runner.os }}
+        
+      - name: download dependencies
+        run: ./gradlew downloadDependencies
+        
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      
+      - name: Build with Gradle
+        run: ./gradlew build dockerBuildImages
+
+      - name: push docker image
```

Comment:
We should not run `dockerPushImages` in context of `pull_request_target`.

Reply:
done